dev_tools/scripts/update_ginkgo_header.sh

```diff
@@ -63,8 +69,10 @@ PREVIOUS_FOLDER=""
 # An empty ${IFS} means the given name (after `read`) will be set to the whole line,
 # and in this case it means it will not ignore leading and trailing whitespaces.
 while IFS='' read -r line; do
-    if [ "${line}" != "${PLACE_HOLDER}" ]; then
-        echo "${line}" >> "${GINKGO_HEADER_TMP}"
+    if [ "$(echo "$line")" != "${PLACE_HOLDER}" ]; then
+        # The line of original file already has the end of line, 
+        # so it does not need to add $END
```

Comment:
By git default, the newline of all files are `\r\n` when you use `git clone`.
When you push it to repo, it will turn `\r\n` to `\n` 
This script uses `\r\n` or `\n` according to `gingko.hpp.in` to keep the same format.